OPHD/Things/Robots/Robot.cpp

```diff
@@ -26,6 +26,12 @@ void Robot::updateTask()
 		die();
 	}
 
+	if (mCancelTask)
+	{
+		mTurnsToCompleteTask = 0;
+		return;
+	}
```

Comment:
Does `mCancelTask` need to be reset after being used?

Edit: Looks like a `reset()` method was added later. Though a `reset()` method would need to be manually called. Maybe it can auto reset once the `mCancelTask` flag is used.

Reply:
I was thinking there had to be a better way to do it and you just nailed it I think. I'll make that adjustment in a separate change set.

Reply:
I'm guessing if the robodozer graphic was simply drawn on top of the tile, mountains might still peek through onto surrounding tiles.

Hmm, actually, it might be interesting to have a set of doze graphics with the mountain gradually wearing down. Would also provide a visual indication of how much doze time was left. That would require graphics work though.

----

Anyway, sounds like there are multiple tasks to cleanup here, and they're not all done at the same time. Maybe code can be refactored so the effects of cancelling can all be done together.

Reply:
The robot graphic effectively is just drawn on top of the terrain tile.

There's the possibility of having robots or structures or even other 'things' that can contain a property that includes a tile to be drawn instead of the tile index. Something like `Image& Thing::terrainTile()` which can then be drawn instead of using the tileset and tile index.

Could allow for interesting effects but something that I want to have a solid grasp on before implementing.